backend/server.js

```diff
@@ -0,0 +1,154 @@
+import dotenv from "dotenv";
+import express from "express";
+import cors from "cors";
+import mongoose from "mongoose";
+
+dotenv.config();
+const port = process.env.PORT || 9000;
+const app = express();
+
+app.use(cors());
+app.use(express.json());
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/thoughtAPI";
+mongoose.connect(mongoUrl, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useCreateIndex: true,
+});
+mongoose.Promise = Promise;
+
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, "Message is required!"],
+    unique: true,
+    trim: true,
+    validate: {
+      validator: (value) => {
+        return /^[^0-9]+$/.test(value);
+      },
+      message: "Numbers are not allowed",
+    },
+    minlength: 5,
+    maxlength: 140,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+});
+
+const Thought = mongoose.model("Thought", thoughtSchema);
+
+app.get("/", (req, res) => {
+  res.send("Hello world");
+});
+
+app.get("/thoughts");
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find();
```

Comment:
The messages supposed to be sorted from the newest.

Reply:
Hey :) Thanks for your comment. It's sorted now.